rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -2246,6 +2260,11 @@ error_code sys_net_bnet_recvfrom(ppu_thread& ppu, s32 s, vm::ptr<void> buf, u32
 
 			native_result = ::recvfrom(sock.socket, reinterpret_cast<char*>(buf.get_ptr()), len, native_flags, reinterpret_cast<struct sockaddr*>(&native_addr), &native_addrlen);
 
+#ifdef _WIN32
+			if(native_result == WSAEMSGSIZE && (native_flags & MSG_PEEK) && len==1)
```

Comment:
native_result is the same on all plateforms, WSAEMSGSIZE is an error value, this makes no sense.

Reply:
You are right of course.
I should have looked for SOCKET_ERROR, and getlasterror to determine the error code.

Reply:
You should scrap received_packet_size, it serves no purpose.
you can check if native_result == -1 && get_native_error() == WSAEMSGSIZE and set native_result to len in that case.
The fact that it's peeking and len is 1 is irrelevant. It's just an "error" to say that the message was truncated and should be ignored.
And if it was truncated ofc the number of bytes read is the length of the buffer passed...

Reply:
Are you sure about the last part?
https://github.com/torvalds/linux/blob/449dc8c97089a6e09fb2dac4d92b1b7ac0eb7c1e/net/socket.c#L2024

It seems to ignore the passed in length for the entire err return path.
It only appears to be used in move_addr_to_user where if it encounters an error then it will output the error.
Otherwise as far as I can see, the value returned is indeed the received data length NOT the passed in buffer length (unless an error occurs... Windows considers truncation an error, Posix doesn't appear to.. and just sets the msghdr field).